src/app/workspace/workspace.service.ts

```diff
@@ -114,4 +115,46 @@ export class WorkspaceService {
     const user = await this.userRepository.findOneBy({ email });
     return { userExists: user ? true : false };
   }
+  
+  public async createWorkspace(createWorkspaceDto: CreateWorkspaceDto, user: User ):
+  Promise<{projectName: string, settings: string, ownerId: number}> {
+    const workspace = { ...createWorkspaceDto, ownerId: user.id};
+    // const workspace = this.workspaceRepository.create(createWorkspaceDto);
+    return this.workspaceRepository.save(workspace);
+  }
+
+  async findAllWorkspaces(): Promise<Workspace[]> {
+    return await this.workspaceRepository.find();
+  }
+
+  async findOneWorkspace(id: number): Promise<Workspace> {
+    const workspace = await this.workspaceRepository.findOneBy({ id });
+    if ( workspace.deletedAt === null ) {
+      throw new Error('Workspace with id ' + { id } + ' was deleted');
+    } else if (!workspace) {
+      throw new Error('Workspace not found');
+    }
+    return workspace;
+  }
+
+  async updateWorkspace(id: number, updateWorkspaceDto: CreateWorkspaceDto) {
+    return await this.workspaceRepository.update(id, updateWorkspaceDto);
```

Comment:
Only owner of the workspace can update and delete workspace and return from update method won't give any useful information to FE.